### PR TITLE
i18nify org.rstudio.studio.client.common

### PR DESCRIPTION
**Pull request by rohank07**
### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any Github issues that are related. 

This PR tackles hard-coded strings to i18n-ify in org.rstudio.studio.client.common (excluding false-positives).
Issue: https://github.com/StatCan/daaas/issues/716

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

This pull-request and future ones will tackle i18nify classes at package-level (e.g. org.rstudio.core) instead of the per-component approach (like we did for [all-options-menu](https://github.com/rstudio/rstudio/pull/9599))

Implementation follows the same pattern from #9485 as described in gwt/tools/i18n-helpers/README.md. Running the create_dev_locale.sh will create all the new "dev" locale from any new "en" properties files for easier visual confirmation. Refer to README for more information.
### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

Modify an entry in one of the new _en.properties files, and refresh RStudio with ant in devmode.
Run create_dev_locale.sh to clone all _en properties files, which will add a prefix of "@" to each constant and message. Then open RStudio using http://localhost:8787/?locale=dev and browse to the Options->* menus. You can also view in desktop mode by swapping the locale element value attribute from value="en" to value="dev" in RStudio.gwt.xml.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

cc @gtritchie

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/common/r/roxygen/RoxygenHelper.java [hunk 1]
```diff
@@ -105,9 +103,9 @@ else if (enclosingScope == "setRefClass")
       else
       {
          globalDisplay_.showErrorMessage(
-             "Insert Roxygen Skeleton",
-             "Unable to insert skeleton (the cursor is not currently " +
-             "inside an R function definition).");
+             constants_.insertRoxygenSkeletonMessage(),
+             constants_.unableToInsertSkeletonMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/r/roxygen/RoxygenHelper.java:107]: Use single string, not two sentence fragments concatenated together.

File: src/gwt/src/org/rstudio/studio/client/common/r/roxygen/RoxygenHelper.java [hunk 2]
```diff
@@ -407,8 +405,8 @@ private void amendExistingRoxygenBlock(
          if (RE_ROXYGEN_NONLOCAL.test(block.get(i)))
          {
             view_.showWarningBar(
-                  "Cannot automatically update roxygen blocks " +
-                  "that are not self-contained.");
+                  constants_.cannotUpdateRoxygenMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/r/roxygen/RoxygenHelper.java:408]: Use single string, not two sentence fragments concatenated together.

File: src/gwt/src/org/rstudio/studio/client/common/rnw/RnwWeaveSelectWidget.java [hunk 3]
```diff
@@ -70,13 +72,13 @@ public void onResponseReceived(TexCapabilities capabilities)
             {
                globalDisplay_.showYesNoMessage(
                   MessageDialog.QUESTION,
-                  "Confirm Change",
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/rnw/RnwWeaveSelectWidget.java:73]: Use a string with placeholders, not a bunch of strings concatenated together.

File: src/gwt/src/org/rstudio/studio/client/common/rpubs/RPubsUploader.java [hunk 4]
```diff
@@ -161,17 +163,17 @@ public void onResponseReceived(Boolean response)
                {
                   onUploadComplete(false);
                   globalDisplay_.showErrorMessage(
-                         "Error",
-                         "Unable to continue " +
-                         "(another publish is currently running)");
+                         constants_.errorCaption(),
+                         constants_.unableToContinueMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/rpubs/RPubsUploader.java:167]: Use single string, not concat.

File: src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java [hunk 5]
```diff
@@ -104,25 +105,25 @@ protected Widget createMainWidget()
       String msg;
       if (!isPublished_ && uploadId_.isEmpty())
       {
-         msg = "RPubs is a free service from RStudio for sharing " +
-                       "documents on the web. Click Publish to get " +
-                       "started.";
+         msg = constants_.rPubsServiceMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java:108]: Use single string.

File: src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java [hunk 6]
```diff
@@ -104,25 +105,25 @@ protected Widget createMainWidget()
...
+                       constants_.clickPublishMessage() +
+                       constants_.startedMessage();
       }
       else
       {
-         msg = "This document has already been published on RPubs. You can " +
-               "choose to either update the existing RPubs document, or " +
-               "create a new one.";
+         msg = constants_.alreadyPublishedRPubs() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java:114]: Single string.

File: src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java [hunk 7]
```diff
@@ -104,25 +105,25 @@ protected Widget createMainWidget()
...
 
       HTML warningLabel =  new HTML(
-        "<strong>IMPORTANT: All documents published to RPubs are " +
-        "publicly visible.</strong> You should " +
-        "only publish documents that you wish to share publicly.");
+        "<strong>" + constants_.importantRPubsMessage() +
+         constants_.publiclyVisibleWarningLabel() + "</strong> " + constants_.youShouldMessage() +
+        constants_.publishDocumentsMessage());
       warningLabel.addStyleName(styles.warningLabel());
       verticalPanel.add(warningLabel);
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java:128]: Use a string with placeholders, not concat, so probably:

```
"<strong>{0}</strong> {1}"
```

File: src/gwt/src/org/rstudio/studio/client/common/rstudioapi/ui/AskSecretDialog.java [hunk 8]
```diff
@@ -113,24 +114,24 @@ public void onKeyDown(KeyDownEvent event)
          );
 
          Label infoLabel = new Label(
-            "Keyring is an R package that provides access to " +
-            "the operating systems credential store to allow you " +
-            "to remember, securely, passwords and secrets. "
+            constants_.keyringDesc() +
+            constants_.keyringCredentialStoreMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/rstudioapi/ui/AskSecretDialog.java:118]: Single string not two

File: src/gwt/src/org/rstudio/studio/client/common/satellite/SatelliteManager.java [hunk 9]
```diff
@@ -132,7 +134,7 @@ public void openSatellite(String name,
       // if we don't need to.
       if (isCurrentWindowSatellite())
       {
-         Debug.log("Satellite windows can't launch other satellites");
+         Debug.log(constants_.satelliteWindowsDebugLog());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/satellite/SatelliteManager.java:137]: Don't localize logging

File: src/gwt/src/org/rstudio/studio/client/common/shell/ShellInteractionManager.java [hunk 10]
```diff
@@ -79,7 +81,7 @@ public void consoleWriteError(String error)
    {
       // show the error in the console then re-prompt
       display_.consoleWriteError(
-            "Error: " + error + "\n");
+            constants_.consoleWriteError() + error + "\n");
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/shell/ShellInteractionManager.java:84]: Single string with placeholders, "Error: {0}\n"

File: src/gwt/src/org/rstudio/studio/client/common/spelling/RealtimeSpellChecker.java [hunk 11]
```diff
@@ -308,4 +309,5 @@ public boolean shouldCheckWord(String word)
 
    private SpellingService spellingService_;
    private UserPrefs userPrefs_;
+   private static final StudioClientCommonConstants constants_ = GWT.create(StudioClientCommonConstants.class);
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/spelling/RealtimeSpellChecker.java:312]: Don't need this, not used.

File: src/gwt/src/org/rstudio/studio/client/common/synctex/model/PdfLocation.java [hunk 12]
```diff
@@ -73,13 +75,14 @@ public final String toDebugString()
    {
       StringBuilder str = new StringBuilder();
       str.append(getFile());
-      str.append("; Page ");
+      str.append(constants_.pdfPageText());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/synctex/model/PdfLocation.java:78]: Part of a debug logging string (`toDebugString()`), don't localize.

File: src/gwt/src/org/rstudio/studio/client/common/synctex/model/PdfLocation.java [hunk 13]
```diff
@@ -73,13 +75,14 @@ public final String toDebugString()
...
       str.append(getPage());
       str.append(" {" + (int)getX() + ", " + (int)getY() + ", " + 
                  (int)getWidth() + ", " + (int)getHeight() + "}");
       if (isFromClick())
-         str.append(" [From Click]");
+         str.append(" " + constants_.pdfFromClickText());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/synctex/model/PdfLocation.java:83]: Debug-only, don't localize

File: src/gwt/src/org/rstudio/studio/client/common/vcs/ShowPublicKeyDialog.java [hunk 14]
```diff
@@ -58,8 +59,8 @@ protected Widget createMainWidget()
       int mod = BrowseCap.hasMetaKey() ? KeyboardShortcut.META : 
                                          KeyboardShortcut.CTRL;
       String cmdText = new KeyCombination("c", 'C', mod).toString(true);
-      HTML label = new HTML("Press " + cmdText + 
-                            " to copy the key to the clipboard");
+      HTML label = new HTML(constants_.pressLabel() + cmdText +
+                            " " + constants_.copyKeyToClipboardLabel());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/vcs/ShowPublicKeyDialog.java:63]: String substitution, not concatenation.

"Press {0} to copy the key to the clipboard"

File: src/gwt/src/org/rstudio/studio/client/common/vcs/ignore/Ignore.java [hunk 15]
```diff
@@ -249,10 +251,10 @@ private IgnoreList createIgnoreList(ArrayList<String> paths,
             GlobalDisplay gDisp = RStudioGinjector.INSTANCE.getGlobalDisplay();
             gDisp.showMessage(
                   MessageDialog.ERROR,
-                  "Error: Multiple Directories",
-                  "The selected files are not all within the same directory " +
-                  "(you can only ignore multiple files in one operation if " +
-                  "they are located within the same directory).");
+                  constants_.multipleDirectoriesCaption(),
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/vcs/ignore/Ignore.java:254]: One string not three.

File: src/gwt/src/org/rstudio/studio/client/common/viewfile/ViewFilePanel.java [hunk 16]
```diff
@@ -356,7 +358,7 @@ public void onUnload()
    private void saveFileAs()
    {
       fileDialogs_.saveFile(
-            "Save File - " + targetFile_.getName(), 
+            constants_.saveFileCaption() + targetFile_.getName(),
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/viewfile/ViewFilePanel.java:361]: Use string substitution, e.g. "Save File - {0}".

**Review by gtritchie — CHANGES_REQUESTED** [review]

File: src/gwt/src/org/rstudio/studio/client/common/satellite/SatelliteManager.java [hunk 17]
```diff
@@ -132,7 +132,7 @@ public void openSatellite(String name,
       // if we don't need to.
       if (isCurrentWindowSatellite())
       {
-         Debug.log("Satellite windows can't launch other satellites");
+         Debug.log("Satellite windows can't launch other satellites");;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/common/satellite/SatelliteManager.java:135]: Remove extra semicolon

**Review by gtritchie — CHANGES_REQUESTED** [review]

File: src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java (hunk 7 above)
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java:122]: Can this be resolved?

**Review by gtritchie — APPROVED** [review]
Looks good, thanks!

**Pull request merged**